docs/csharp/language-reference/keywords/lock-statement.md

```diff
@@ -11,76 +11,69 @@ ms.assetid: 656da1a4-707e-4ef6-9c6e-6d13b646af42
 ---
 # lock statement (C# Reference)
 
-The `lock` keyword marks a statement block as a critical section by obtaining the mutual-exclusion lock for a given object, executing a statement, and then releasing the lock. The following example includes a `lock` statement.
+The `lock` statement obtains the mutual-exclusion lock for a given object, executes a statement block, and then releases the lock.
+
+The `lock` statement of the form
 
 ```csharp
-class Account
+lock (x)
 {
-    decimal balance;
-    private Object thisLock = new Object();
-
-    public void Withdraw(decimal amount)
-    {
-        lock (thisLock)
-        {
-            if (amount > balance)
-            {
-                throw new Exception("Insufficient funds");
-            }
-            balance -= amount;
-        }
-    }
+    // Your code...
 }
 ```
 
-For more information, see [Thread Synchronization](../../programming-guide/concepts/threading/thread-synchronization.md).
-
-## Remarks
-
-The `lock` keyword ensures that one thread does not enter a critical section of code while another thread is in the critical section. If another thread tries to enter a locked code, it will wait, block, until the object is released.
+where `x` is an expression of a [reference type](reference-types.md) is precisely equivalent to
 
-The section [Threading](../../programming-guide/concepts/threading/index.md) discusses threading.
+```csharp
+bool __lockWasTaken = false;
+try
+{
+    System.Threading.Monitor.Enter(x, ref __lockWasTaken);
+    // Your code...
+}
+finally
+{
+    if (__lockWasTaken) System.Threading.Monitor.Exit(x);
+}
+```
 
-The `lock` keyword calls <xref:System.Threading.Monitor.Enter%2A> at the start of the block and <xref:System.Threading.Monitor.Exit%2A> at the end of the block. A <xref:System.Threading.ThreadInterruptedException> is thrown if <xref:System.Threading.Thread.Interrupt%2A> interrupts a thread that is waiting to enter a `lock` statement.
+except that `x` is only evaluated once.
```

Comment:
Might be better to just show that in the example - add `object __lockObj = x;` and use `__lockObj` in the monitor calls.